src/tests/array/test_logicalloc.f90

```diff
@@ -0,0 +1,154 @@
+! SPDX-Identifier: MIT
+
+module test_logicalloc
+  use stdlib_array, only : trueloc, falseloc
+  use stdlib_string_type, only : string_type, len
+  use testdrive, only : new_unittest, unittest_type, error_type, check
+  implicit none
+  private
+
+  public :: collect_logicalloc
+
+contains
+
+  !> Collect all exported unit tests
+  subroutine collect_logicalloc(testsuite)
+    !> Collection of tests
+    type(unittest_type), allocatable, intent(out) :: testsuite(:)
+
+    testsuite = [ &
+      new_unittest("trueloc-where", test_trueloc_where), &
+      new_unittest("trueloc-merge", test_trueloc_merge), &
+      new_unittest("falseloc-where", test_falseloc_where), &
+      new_unittest("falseloc-merge", test_falseloc_merge) &
+      ]
+  end subroutine collect_logicalloc
+
+  subroutine test_trueloc_where(error)
+    !> Error handling
+    type(error_type), allocatable, intent(out) :: error
+
+    integer :: ndim
+    real, allocatable :: avec(:), bvec(:), cvec(:)
+
```

Comment:
Good for the tests to include some corner cases, such as `size(array) == 0`, and also `array(:) = .true.` and `array(:) = .false.`, for both `trueloc` and `falseloc` (unless they do not support that, in which case just document it in the specs).